examples/rust/files/chunk_sender/src/main.rs

```diff
@@ -0,0 +1,638 @@
+            .client
+            .aqc()
+            .create_bidi_channel(team_id, receiver_net_identifier, file_transfer_label)
+            .await;
+        (channel_result, sender)
+    });
+
+    info!("receiver receiving aqc bidi channel");
+    let AqcPeerChannel::Bidi(mut received_aqc_chan) =
+        receiver.client.aqc().receive_channel().await?
+    else {
+        bail!("expected a bidirectional channel");
+    };
+
+    let (created_aqc_chan_result, sender_returned) = create_handle
+        .await
+        .context("Task for sender creating bidi channel panicked")?;
+    sender = sender_returned;
+    let mut created_aqc_chan =
+        created_aqc_chan_result.context("Sender failed to create bidi channel")?;
+
+    // Use the new multi-stream approach
+    info!("sender sending file over multiple streams");
+    send_file_over_multiple_streams(&file_path, &mut created_aqc_chan, chunk_size).await?;
+
+    // receiver receives data from multiple streams
+    info!("receiver receiving file data from multiple streams");
+    let output_filename = file_path.file_name()
+        .context("input file has no filename")?
+        .to_str()
+        .context("filename is not valid UTF-8")?;
+    let output_path = Path::new("received_").join(output_filename);
+    if let Some(parent) = output_path.parent() {
+        fs::create_dir_all(parent).await
+            .with_context(|| format!("failed to create directory: {:?}", parent))?;
+    }
+    
+    receive_file_from_multiple_streams(&output_path, chunk_size, &mut received_aqc_chan).await?;
+    info!("file saved successfully: {:?}", output_path);
+
+    info!("revoking label from sender");
+    operator_team.revoke_label(sender.id, file_transfer_label).await?;
+    info!("revoking label from receiver");
+    operator_team.revoke_label(receiver.id, file_transfer_label).await?;
+    info!("deleting label");
+    admin_team.delete_label(file_transfer_label).await?;
+
+    info!("completed file transfer demo");
+
+    info!("completed Aranya file transfer application");
+
+    Ok(())
+}
+
+
+
+fn optimal_chunk_size(file_size: usize) -> usize {
+    // Maximum available memory is 1GB, so max chunk size is 1/15 of 1GB to leave room for system and other services
+    let max_chunk_size = (1024 * 1024 * 1024) / 15; // ~68 MB
+    let min_chunk = 8 * 2; // 16 KiB
+    
+    // Ensure at least 2 chunks by limiting chunk size to half the file size
+    let max_chunk_for_min_2 = file_size / 2;
+    
+    // Use the smaller of: max_chunk_size, max_chunk_for_min_2, or min_chunk
+    let chunk_size = max_chunk_size.min(max_chunk_for_min_2).max(min_chunk);
+    
+    // If the calculated chunk size would result in only 1 chunk, force it smaller
+    if file_size <= chunk_size {
+        chunk_size / 2
+    } else {
+        chunk_size
+    }
+}
+
+async fn send_file_over_multiple_streams(
+    file_path: &Path,
+    channel: &mut AqcBidiChannel,
+    chunk_size: usize,
+) -> Result<()> {
```

Comment:
The chunk size can be computed within this function:
```suggestion
async fn send_file_over_multiple_streams(
    file_path: &Path,
    channel: &mut AqcBidiChannel,
) -> Result<()> {
```